tests/test_utils/test_transactiontestcase.py

```diff
@@ -26,3 +45,22 @@ def test(self, call_command):
             reset_sequences=False, inhibit_post_migrate=True,
             database='default', verbosity=0,
         )
+
+
+class TestDataConsistencyOnTearDown(TestSerializedContentMockMixin, TransactionTestCase):
+    """
+    TransactionTestCase._fixture_teardown() is flushing the whole database.
+    If an other test after this one needs some data coming from initial data migrations,
+    there data won't be found.
+    One solution is to load again initial data migrations after the flush called on _fixture_teardown().
+    """
+    available_apps = ['test_utils']
+    serialized_rollback = True
+    initial_data_migration = '[{"model": "test_utils.car", "pk": 1, "fields": {"name": "K 2000"}}]'
+
+    def _post_teardown(self):
+        super(TestDataConsistencyOnTearDown, self)._post_teardown()
+        self.assertTrue(Car.objects.exists())
```

Comment:
The problem is what are we defining as `clean state`. For me, the `clean state` is the state where the database is after the initial data migration(s) and before the test suite runs. 

When you apply initial migrations, they can for example initialise let's say some permissions.

You then want each test to be launched with these initial data migrations. The permissions created initially should be in each test, which means that when we flush the db on `TransactionTestCase`, we have to get our initial permissions back, so that the next `TransactionTestCase` will run with these permissions.

And more than that, with `--keepdb` option, you need your test database to still contain these permissions (they won't be initialised anymore because your db has kept `django_migrations` history and won't recreate initial data migrations), or your tests will probably fail next time you run them.